File: 프로그래머스/unrated/161989. 덧칠하기/덧칠하기.py
from collections import deque

def solution(n, m, section):
    answer = 0
    section = deque(section)
    
    while section:
        left = section.popleft()
        while section and section[0]-left+1 <= m:
            right = section.popleft()
        answer += 1
    return answer

File: 프로그래머스/unrated/161989. 덧칠하기/덧칠하기.java
import java.util.*;

class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int temp;
        Deque<Integer> q = new ArrayDeque();
        
        for (int i: section)
            q.addLast(i);
        
        while (!q.isEmpty()) {
            temp = q.removeFirst();
            answer++;
            while (!q.isEmpty() && q.peekFirst() - temp + 1 <= m)
                q.removeFirst();
        }
        
        return answer;
    }
}